radiotools/coreas/generate_coreas_sim.py

```diff
@@ -242,21 +408,98 @@ def write_list_star_pattern(filename, zen, az, append=False, obs_level=1564.0, o
                 slices = np.array(slices)
                 if(slicing_method == "distance"):
                     slices *= 100
-                for iSlice in xrange(len(slices) - 1):
-                    name = "pos_%i_%i_slice%i" % (rs[i], np.rad2deg(azimuths[j]), iSlice)
+                for iSlice in range(len(slices) - 1):
+                    name = "pos_%i_%i_slice%i" % (antenna_rings[i], np.rad2deg(arm_orientiations[j]), iSlice)
                     if gammacut is None:
-                        fout.write('AntennaPosition = {0} {1} {2} {3} {4} {5} {6}\n'.format(x, y, z, name, slicing_method, slices[iSlice] * 100., slices[iSlice + 1] * 100.))
+                        # save the generated starshapes to the antenna list file
+                        # positions in cm
+                        fout.write('AntennaPosition = {0} {1} {2} {3} {4} {5} {6}\n'.format(antennas[0], antennas[1], antennas[2], name, slicing_method, slices[iSlice] * 100., slices[iSlice + 1] * 100.))
                     else:
                         for iG, gcut in enumerate(gammacut):
                             name_gamma = "%s_gamma%i" % (name, iG)
-                            fout.write('AntennaPosition = {0} {1} {2} {3} {4} {5} {6} gamma {7} {8}\n'.format(x, y, z, name_gamma, slicing_method, slices[iSlice] * 100., slices[iSlice + 1] * 100., gcut[0], gcut[1]))
+                            # save the generated starshapes to the antenna list file
+                            # positions in cm
+                            fout.write('AntennaPosition = {0} {1} {2} {3} {4} {5} {6} gamma {7} {8}\n'.format(antennas[0], antennas[1], antennas[2], name_gamma, slicing_method, slices[iSlice] * 100., slices[iSlice + 1] * 100., gcut[0], gcut[1]))
+    
+    print("Saved antenna positions (in groundplane coordinates) to file: ", filename)
+
     fout.close()
 
+    # in case you want to plot the antennas in the shower plane coordinate system
+    # mainly for visual checking whether the starshape is ok
+    if vxB_plot==True:
+        # open the shower.list file to save the generated starshapes to
+        with open("shower_plane.list", "w") as file:
+                
+            # transform the station positions to vxB system for plot
+            shower_plane_system = cs.transform_to_vxB_vxvxB(np.array(station_positions_groundsystem))
+         
+            for i in range(len(shower_plane_system)):
+                # save the generated starshapes to the antenna.list file
+                # positions in cm
+                file.write(f"AntennaPosition = {shower_plane_system[i, 0]} {shower_plane_system[i, 1]} {shower_plane_system[i, 2]} {name}\n")
+            
+            print("Saved antenna positions (in vxB_vxvxB coordinates) to file: ", "shower_plane.list")
+
+
+    # return corsika azimuth angle to for automatically generating corsika input files with the right values
+    return corsika_azimuth
+
+
+def get_rmax(X):
```

Comment:
The input for the function is already observation level dependent since it uses the grammage at observation level when it's used in the next function . The grammage is calculated with "get_atmospherere" from the models.py file.
In any case, it needs more documentation.

I agree it's not great solution for automatically generating antenna rings. When using this branch to generate simulations recently, I also noticed that the new function "get_starshaped_pattern_radii" that I added has a dependency to a function from @fschlueter's RadioAnalysis framework to estimate the Cherenkov radius of a shower. 
I used these two functions to, independly from geometry, generate antenna rings that go to big axis distances, but have a higher density from the shower core to a little further out than the Cherenkov radius.
I think it's an important functionality to have, but I might have to implement it differently. If I kept it mostly the same, then I need to implement the Cherenkov radius functions into radiotools as well.